src/metabase/api/dashboard.clj

```diff
@@ -397,16 +399,22 @@
                                                existing-mappings (get dashcard-id->existing-mappings dashcard-id)]
                                            (contains? existing-mappings (select-keys mapping [:target :parameter_id]))))
         new-mappings                   (for [{mappings :parameter_mappings, dashcard-id :id} dashcards
-                                             mapping                                         mappings
-                                             :when                                           (not (existing-mapping? dashcard-id mapping))]
+                                             mapping mappings
+                                             :when (not (existing-mapping? dashcard-id mapping))]
                                          (assoc mapping :dashcard-id dashcard-id))
         ;; need to add the appropriate `:card-id` for all the new mappings we're going to check.
         dashcard-id->card-id           (when (seq new-mappings)
                                          (db/select-id->field :card_id DashboardCard
                                            :dashboard_id dashboard-id
                                            :id           [:in (set (map :dashcard-id new-mappings))]))
-        new-mappings (for [{:keys [dashcard-id], :as mapping} new-mappings]
-                       (assoc mapping :card-id (get dashcard-id->card-id dashcard-id)))]
+        new-mappings                   (reduce (fn [new-mappings {:keys [dashcard-id] :as mapping}]
+                                                 ;; nil card-id means the DashboardCard is a text card, so we don't
+                                                 ;; need to check permissions
+                                                 (if-let [card-id (get dashcard-id->card-id dashcard-id)]
```

Comment:
a dashcard without card-id is a text card seems to be a bit loose right?
Could we check the `virtual_card.display == "text"` instead?

Reply:
Actually, I think `check-parameter-mapping-permissions` should just skip permission checks for dashcards with no `card-id` in general. If there's no card ID, there's no query, and it thus doesn't make sense to do a permission check.

With that function changed, I can revert the lines here.